### PR TITLE
Add user/team participation

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/Challenge.java [hunk 1]
```diff
@@ -97,12 +93,12 @@ public void setInviteOnly(boolean inviteOnly) {
         this.inviteOnly = inviteOnly;
     }
 
-    public Set<User> getRegisteredUsers() {
-        return registeredUsers;
+    public List<Participation> getParticipations() {
```
Comment (flowlo) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/Challenge.java]: Should be a `Set`.

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/Challenge.java [hunk 2]
```diff
@@ -57,6 +50,9 @@
     @Column(name = "end_date")
     private ZonedDateTime endDate;
 
+    @OneToMany(mappedBy = "challenge")
+    private List<Participation> participations;
```
Comment (flowlo) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/Challenge.java]: Should be a `Set`.

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/User.java [hunk 3]
```diff
@@ -88,6 +85,9 @@
     @Column(name = "last_login")
     private ZonedDateTime lastLogin;
 
+    @OneToMany(mappedBy = "user")
+    private List<Participation> participations;
```
Comment (flowlo) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/User.java]: Should probably be a `Set` as well, or is that list ordered by date or something?

Comment (victorbalan) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/User.java]: no..i`ll make it a set everywhere

**Pull request closed without merging**